### ISSUE TITLE
grpcWeb Transport: Timeout

### ISSUE THREAD

**Issue by pzeinlinger** [timostamm/protobuf-ts#99]
Since #80 the deadline property is treated as a proper deadline.
But have you thought about an option to specify a per-request timeout?

**Comment by timostamm** [comment]
IMO you probably want to specify a timeout, most of the time. So I think there should be an option to set a timeout, both per-request and as a default for the transport. 

Could look like this:

```ts
    /**
     * Timeout for the call in milliseconds.
     */
    timeout?: number;
```

But what happens when the user specifies both `timeout` and `deadline`? 


Alternative: Remove the "deadline" property, add a "timeout" property that accepts numbers as timeouts in ms or deadlines as Date:

```ts
    /**
     * Timeout for the call in milliseconds.  
     * If a Date object is given, it is used as a deadline.
     */
    timeout?: number | Date;
```

This is easy to resolve in transport implementations and works well with merging default / per-request options.

Breaking change is acceptable, 2.0 is not released yet.

Not changed yet because testing takes time and #57, #56 have higher prio for me right now.

Alternatives or PR are welcome.

**Comment by timostamm** [comment]
Gave the new implementation a proper test with [example-dotnet-grpcweb-server](https://github.com/timostamm/protobuf-ts/tree/master/packages/example-dotnet-grpcweb-server) and [example-node-grpc-transport-client](https://github.com/timostamm/protobuf-ts/tree/master/packages/example-node-grpc-transport-client).

Passed deadlines / negative timeouts are now caught client-side (the spec doesn't allow negative values, so behavior is undefined).

**Comment by timostamm** [comment]
Released in v2.0.0-alpha.30 in the "next" channel.
